alica_engine/include/engine/constraintmodul/Query.h

```diff
@@ -132,12 +132,13 @@ bool Query::getSolution(ThreadSafePlanInterface pi, std::vector<ResultType>& res
     // Collect the complete problem specification
     std::vector<std::shared_ptr<ProblemDescriptor>> cds;
     int domOffset;
-    SolverType* solver = pi.getAlicaEngine()->getSolver<SolverType>();
-    if (solver == nullptr) {
+
+    if (!pi.getAlicaEngine()->existSolver<SolverType>()) {
         ALICA_ERROR_MSG("Query::getSolution: The engine does not have a suitable solver for the given type available.");
         return false;
     }
 
+    SolverType& solver = pi.getAlicaEngine()->getSolver<SolverType>();
```

Comment:
Double lookup. Don't look up twice

Reply:
For that to happen, I don't see any other option but to return pointer of solver rather than reference. Thought of following the STL model here. We can also move exist() checking to autogenerated code, ideally, app should be aware of what solvers were added and won't need the exist() checking.